calendar.rb

```diff
@@ -0,0 +1,115 @@
+require 'date'
+require_relative 'event'
+require_relative 'home'
+
+
+class Calendar
+  attr_accessor :events
+  def initialize
+    @events = Hash.new { |h, k| h[k] = [] }
+    load_data
+  end
+  
+
+  def load_data
+    file_data = File.open("events.txt");
+    File.foreach("events.txt") do |line|
+      date_from_file, details_from_file = line.split(',')
+      event_data = Event.new(Date.parse(date_from_file),details_from_file.chomp)
+      @events[event_data.date.month].push(event_data)
+    end
+  end
+
+
+  #this method returns the matched events line numbers in an array
+  def get_line_number(file, date)
+    return "incorrect date entered" unless date
+    count = 0
+    arr = []
+    file = File.open(file, "r") { |file| file.each_line { |line|
+      arr << count if line =~ /#{date}/
+      count += 1
+      }
+    }
+    arr
+  end
+
+  def add_event(date,details)
+    event_to_add = Event.new(date,details)
+    @events[date.month].push(event_to_add)
+    # puts @events
+    File.open("events.txt", "a") { |f| f.write "#{event_to_add.date},#{event_to_add.details}\n" }
+    "Successfully added an event"
+  end
+
+  #this method displays events of specified date
+  def view_events(date)
+    verify = File.read("events.txt").include?(date.to_s)
```

Comment:
verify variable name is too generic. Variable names should be intention revealing.